sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/GeneralScalarExpression.java

```diff
@@ -22,20 +22,46 @@
 import org.apache.spark.annotation.Evolving;
 
 /**
- * The general SQL string corresponding to expression.
+ * The general V2 expression corresponding to V1 expression.
+ * <p>
+ * The currently supported expression:
+ * <ol>
+ *  <li><pre>CASE WHEN expr1 THEN expr2 [WHEN expr3 THEN expr4]* [ELSE expr5] END</pre>
+ *  Since 3.3.0</li>
+ * </ol>
  *
  * @since 3.3.0
  */
 @Evolving
-public class GeneralSQLExpression implements Expression, Serializable {
-    private String sql;
+public class GeneralScalarExpression implements Expression, Serializable {
+  private String name;
+  private Expression[] children;
+  private String sql;
```

Comment:
why do we need this field?

Reply:
I want PushedAggregates keeps the SQL.
```
        val expected_plan_fragment =
          "PushedAggregates: [COUNT(CASE WHEN ((SALARY) > (8000.00)) AND ((SALARY) < (10000.00))" +
            " THEN SALARY ELSE 0.00 END), C..., " +
            "PushedFilters: [], " +
            "PushedGroupByColumns: [DEPT]"
```

Reply:
setting SQL string is pretty ugly. Can't we just implement the `sql` method in this class with `V2ExpressionSQLBuilder`?

Reply:
OK